website/docs/runtime.md

```diff
@@ -101,22 +100,150 @@ etc.) or a completely untyped language like Ruby; a
-available and how to register them.
+- **Whether** the runtime checks run in the first place.
+
+  To change this, we use `.checked` in a method signature.
+
+In the next sections, we'll give some examples of how to use both.
+
+## `.on_failure`: Changing what happens on runtime errors
+
+By adding `.on_failure(...)` to a sig and registering a `T::Configuration`
+callback, we can change what happens when a sig check fails. For example:
+
+```ruby
+require 'sorbet-runtime'
+
+# (1) Register call_validation_error_handler callback.
+# This runs every time a method with a sig fails to type check at runtime.
+T::Configuration.call_validation_error_handler = lambda do |signature, opts|
+  #                                                     ┌──┐
+  if signature.on_failure && signature.on_failure[0] == :log
+    puts opts[:pretty_message]
+  else
+    raise TypeError.new(opts[:pretty_message])
+  end
+end
+
+class Main
+  extend T::Sig
+
+  # (2) Use .on_failure in the sig for a method
+  #                                       ┌───────────────┐
+  sig {params(argv: T::Array[String]).void.on_failure(:log)}
+  def self.main(argv)
+    puts argv
+  end
+end
+
+# (3) When we call main incorrectly, it will print
+# with puts instead of raise an exception.
+Main.main(42)
+
+# Output:
+# ❯ ruby example.rb
+# Parameter 'argv': Expected type T::Array[String], got type Integer with value 42
+# Caller: example.rb:25
+# Definition: example.rb:18
+# 42
+```
+
+We defined our own meaning for `.on_failure` with `T::Configuration`. Without
+doing this, `.on_failure` has no effect. Because of this, `.on_failure` can be
+completely customized within any codebase to change what it means to fail. For
+example at Stripe, we use `.on_failure` to attach team ownership information to
+a failure.
+
+The `T::Configuration` handler we wrote branches on whether `.on_failure` was
+provided, which means the logging behavior is opt in. This is nice because it
+means the default behavior is still to make problems with types fail loudly and
+early, rather than silently as a log. If we wanted, we could have inverted this:
+
+```ruby
+T::Configuration.call_validation_error_handler = lambda do |signature, opts|
+  #                                                     ┌────┐
+  if signature.on_failure && signature.on_failure[0] == :raise
+    raise TypeError.new(opts[:pretty_message])
+  else
+    puts opts[:pretty_message]
+  end
+end
+
+# ...
+
+#                                       ┌─────────────────┐
+sig {params(argv: T::Array[String]).void.on_failure(:raise)}
+```
+
+With this `T::Configuration` handler, the default is to log, and we can use
+`.on_failure` to opt specific sigs into raising on failure.
+
+For more on the various `T::Configuration` handlers, see
+[Runtime Configuration](tconfiguration.md).
+
+## `.checked`: Whether to check in the first place
+
+> **Careful!** Opting out of runtime checks opts out of significant safety
```

Comment:
I think this is a little technical and vague to someone who don't already know what the consequences are. I would say something like `Opting out of runtime checks can significantly degrade the trustworthiness of type signatures. Only
disable the runtime when you're aware of the tradeoffs.` to be more visceral. Maybe also link to https://sorbet.org/docs/gradual for someone who wants to understand the trade-offs?

Reply:
It might be worth noting (possibly here, possibly elsewhere) the kind of failure cases that arise from turning off checking, and in particular the possibility that silently violating a type contract in one place can cause errors to arise in a different place entirely.